tests/scripts/thread-cert/Cert_5_1_03_RouterAddressReallocation.py

```diff
@@ -72,113 +95,133 @@ def test(self):
-        msg.assertCoapMessageContainsTlv(network_layer.Status)
-
-        # 8 - Router2
-        msg = router2_messages.next_coap_message("2.04")
+        self.collect_rloc16s()
+
+    def verify(self, pv):
+        pkts = pv.pkts
+        pv.summary.show()
+
+        LEADER = pv.vars['LEADER']
+        ROUTER_1 = pv.vars['ROUTER_1']
+        ROUTER_2 = pv.vars['ROUTER_2']
+        ROUTER_2_RLOC16 = pv.vars['ROUTER_2_RLOC16']
+
+        # Step 2: Verify topology is formed correctly.
+        pv.verify_attached('ROUTER_1')
+        pv.verify_attached('ROUTER_2')
+        _pkt = pkts.filter_wpan_src64(ROUTER_2).\
+                   filter_mle_cmd(MLE_ADVERTISEMENT).\
+                   must_next()
+
+        # Step 5: Router_1 MUST attempt to reattach to its original partition
+        #         by sending a MLE Parent Request with a hop limit of 255 to
+        #         the All-Routers multicast address (FF02::2).
+        #         The following TLVs MUST be present in the MLE Parent Request:
+        #            - Challenge TLV
+        #            - Mode TLV
+        #            - Scan Mask TLV (MUST have E and R flags set)
+        #            - Version TLV
+        #         The DUT MUST make two separate attempts to reconnect to its
+        #         original partition in this manner
+
+        for i in range(1, 3):
+            pkts.filter_wpan_src64(ROUTER_1).\
+                filter_LLARMA().\
+                filter_mle_cmd(MLE_PARENT_REQUEST).\
+                filter(lambda p: {
+                                  CHALLENGE_TLV,
+                                  MODE_TLV,
+                                  SCAN_MASK_TLV,
+                                  VERSION_TLV
+                                  } <= set(p.mle.tlv.type) and\
+                       p.ipv6.hlim == 255 and\
+                       p.mle.tlv.scan_mask.r == 1 and\
+                       p.mle.tlv.scan_mask.e == 1).\
+                must_next()
+
+        # Step 6: Router_1 MUST attempt to attach to any other partition
+        #         within range by sending a MLE Parent Request.
+        #         The following TLVs MUST be present in the MLE Parent Request:
+        #            - Challenge TLV
+        #            - Mode TLV
+        #            - Scan Mask TLV
+        #            - Version TLV
+
+        pkts.filter_wpan_src64(ROUTER_2).\
+                   filter_mle_cmd(MLE_ADVERTISEMENT).\
+                   must_next().\
+                   must_verify(lambda p:\
+                               p.mle.tlv.leader_data.partition_id !=\
+                               _pkt.mle.tlv.leader_data.partition_id)
+
+        pkts.filter_wpan_src64(ROUTER_1).\
+            filter_LLARMA().\
+            filter_mle_cmd(MLE_PARENT_REQUEST).\
+            filter(lambda p: {
+                              CHALLENGE_TLV,
+                              MODE_TLV,
+                              SCAN_MASK_TLV,
```

Comment:
scan mask tlv in first parent request when attach any should be `R` only?

Reply:
added the check

Reply:
p.mle.tlv.scan_mask.r == 0 should be added as well?

Reply:
added